docs/style.md

```diff
@@ -196,6 +196,53 @@ external crates. The surprise factor for derivations sharing a name with the sta
 library traits (`Display`) is reduced and it also acts as natural mechanism to tell apart names
 prone to collision (`Serialize`), all without needing to look up the list of imports.
 
+### Visibility
+
+"Minimize visibility" is a standard best practice, but rust has some extra
+specifics. In Rust, there are two granularities for visibility: visibility of
+items within a single crate, and visibility of items across the crates.
+Cross-crate APIs often form important architectural boundaries affecting the
+whole application, while what happens inside a crate concerns only the crate
+itself.
+
+The difference between `pub` (making items visible to other crates) and
+`pub(crate)` (making items visible within the crate) is big, while the
+difference between `pub(crate)` and private is relatively small. Be mindful when
+exposing APIs across crates with `pub`.
+
+For a top-level item, use `pub` only when it must be reachable from outside of
+the crate, don't rely on effective visibility:
+
+```rust
+// GOOD
+pub(crate) mod auth {
+    pub(crate) check_password(password: &str) -> Result<(), ()>
+}
+
+// BAD
+pub(crate) mod auth {
+    pub check_password(password: &str) -> Result<(), ()>
+}
+```
+
+**Rationale:** it's useful to tell at a glance that `pub` items form crate's
+public API without checking all the ancestor modules.
+
+For fields and methods, it's OK to use `pub` even if the type itself is not
```

Comment:
I would mark this somehow that this text is just a note, clarification, rather than a part of the rationale, given the structure of the rest of the document.

Reply:
I think it's clear enough from context. Rationale is always a single paragraph, and we do use that in, eg, https://github.com/near/nearcore/blob/c1ae98f1bc1c9978ca7f6399fedc9542649558cc/docs/style.md#avoid-references-to-copy-types